pandas/core/dtypes/dtypes.py

```diff
@@ -811,6 +811,50 @@ def __eq__(self, other: Any) -> bool:
             and tz_compare(self.tz, other.tz)
         )
 
+    def __from_arrow__(
+        self, array: pyarrow.Array | pyarrow.ChunkedArray
+    ) -> DatetimeArray:
+        """
```

Comment:
why is this method on the dtype? we don't construct others this way

Reply:
That’s how it’s documented here: https://pandas.pydata.org/docs/development/extending.html?highlight=__from_arrow__#compatibility-with-apache-arrow

I’m not 100% sure why the array goes to arrow, but the dtype goes from arrow. Maybe because the dtype is more of a public interface?

Reply:
Some examples:

* NumericDtype https://github.com/pandas-dev/pandas/blob/f8a37a76c5d8f22cc9937e08c0c9555e75f819a4/pandas/core/arrays/numeric.py#L64
* ArrowDtype https://github.com/pandas-dev/pandas/blob/52c653b437db67815203872de6179b1f7adeb148/pandas/core/arrays/arrow/dtype.py#L317
* BooleanDtype https://github.com/pandas-dev/pandas/blob/52c653b437db67815203872de6179b1f7adeb148/pandas/core/arrays/boolean.py#L103
* StringDtype https://github.com/pandas-dev/pandas/blob/57c49405f2bd31f797991f2a36dfa0036cb5adfe/pandas/core/arrays/string_.py#L185
* PeriodDtype https://github.com/pandas-dev/pandas/blob/b1c6622bcfc0093ee1490d581dad724c431e1ea2/pandas/core/dtypes/dtypes.py#L1002
* IntervalDtype https://github.com/pandas-dev/pandas/blob/b1c6622bcfc0093ee1490d581dad724c431e1ea2/pandas/core/dtypes/dtypes.py#L1254